uobtheatre/reports/schema.py

```diff
@@ -33,6 +33,22 @@ class ReportOption(graphene.InputObjectType):
     value = graphene.String(required=True)
 
 
+class DataSetNode(graphene.ObjectType):
+    name = graphene.String(required=True)
+    headings = graphene.List(graphene.String, required=True)
+    data = graphene.List(graphene.List(graphene.String), required=True)
+
+
+class MetaItemNode(graphene.ObjectType):
+    name = graphene.String(required=True)
+    value = graphene.String(required=True)
+
+
+class ReportNode(graphene.ObjectType):
+    datasets = graphene.List(DataSetNode)
+    meta = graphene.List(MetaItemNode)
+
+
 class GenerateReport(AuthRequiredMixin, SafeMutation):
```

Comment:
Should this not be a query? It feels like this is not mutating any data but just getting something

Reply:
I think it should stay a mutation because it is doing a (fairly expensive) operation by collating the data. This can include calls to Square, etc. This is much more a complex algorithm compared to a simply property on a model (which is essentially what querying is for) - you are executing a function here that can actually do work and update / change things - the definition of a mutaiton.